framework/PostProcessors/ExternalPostProcessor.py

```diff
@@ -97,7 +97,7 @@ def inputToInternal(self, currentInput):
               + " but multiple inputs are provided!")
       else:
         currentInput = currentInput[-1]
-    assert(hasattr(currentInput, 'type'), "The type is missing for input object! We should always associate a type with it.")
+    assert(hasattr(currentInput, 'type')), self.raiseAnError(IOError, "The type is missing for input object! We should always associate a type with it.")
```

Comment:
I would not use the message handler here since you did not use it in any other place

Reply:
fixed